extension/storage/local/filestorage/client.go

```diff
@@ -0,0 +1,96 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package filestorage
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"go.etcd.io/bbolt"
+)
+
+var defaultBucket = []byte(`default`)
+
+type fileStorageClient struct {
+	db *bbolt.DB
+}
+
+func newClient(file string, timeout time.Duration) (*fileStorageClient, error) {
+	options := &bbolt.Options{Timeout: timeout}
+	db, err := bbolt.Open(file, 0666, options)
+	if err != nil {
+		return nil, err
+	}
+
+	initBucket := func(tx *bbolt.Tx) error {
+		_, err := tx.CreateBucketIfNotExists(defaultBucket)
+		return err
+	}
+	if err := db.Update(initBucket); err != nil {
+		return nil, err
+	}
+
+	return &fileStorageClient{db}, nil
+}
+
+// Get will retrieve data from storage that corresponds to the specified key
+func (c *fileStorageClient) Get(_ context.Context, key string) ([]byte, error) {
+	var result []byte
+	get := func(tx *bbolt.Tx) error {
+		bucket := tx.Bucket(defaultBucket)
+		if bucket == nil {
+			return fmt.Errorf("storage not initialized")
```

Comment:
Nit (I think our linter wants this):
```suggestion
			return errors.New("storage not initialized")
```